connector/google/google.go

```diff
@@ -257,12 +258,15 @@ func (c *googleConnector) getGroups(email string, fetchTransitiveGroupMembership
 		}
 
 		for _, group := range groupsList.Groups {
-			// TODO (joelspeed): Make desired group key configurable
-			userGroups = append(userGroups, group.Email)
+			if _, exists := (*checkedGroups)[group.Email]; !exists {
+				(*checkedGroups)[group.Email] = struct{}{}
+				// TODO (joelspeed): Make desired group key configurable
+				userGroups = append(userGroups, group.Email)
+			}
 
 			// getGroups takes a user's email/alias as well as a group's email/alias
 			if fetchTransitiveGroupMembership {
-				transitiveGroups, err := c.getGroups(group.Email, fetchTransitiveGroupMembership)
+				transitiveGroups, err := c.getGroups(group.Email, fetchTransitiveGroupMembership, checkedGroups)
```

Comment:
Do we need to check transitive group membership if a groups is already in checked groups?

Reply:
I'm unsure if I understand the question, but if you are asking if `fetchTransitiveGroupMembership` should always be `false` here then I think the answer is no due to multiple levels of nesting.

Reply:
I see what you were saying now. I've moved this block inside of the if condition as well.

Reply:
Done. Same question as above on who resolves